models.py

```diff
@@ -1051,17 +1081,32 @@ class FundingRecord(BaseModel, AuditMixin):
     disambiguated_org_identifier = CharField(null=True, max_length=255)
     disambiguation_source = CharField(null=True, max_length=255)
     visibility = CharField(null=True, max_length=100)
-    put_code = IntegerField(null=True)
+    is_active = BooleanField(
+        default=False, help_text="The record is marked for batch processing", null=True)
+    processed_at = DateTimeField(null=True)
+    status = TextField(null=True, help_text="Record processing status.")
 
     @classmethod
     def load_from_json(cls, source, filename=None, org=None):
         """Load data from CSV file or a string."""
         if isinstance(source, str):
-            funding_data = json.loads(source)
+            # import data from file based on its extension; either it is yaml or json
+            if os.path.splitext(filename)[1][1:] == "yaml" or "yml":
+                funding_data = yaml.load(source)
```

Comment:
TODO: since YAML expected to be much simpler (excluded redundant hierarchy level) it needs to be mapped back to ORCID schema.